new_client/src/actions/authActions.js

```diff
@@ -0,0 +1,100 @@
+import {
+  USER_LOADED,
+  USER_LOADING,
+  AUTH_ERROR,
+  LOGIN_SUCCESS,
+  LOGIN_FAIL,
+  REGISTER_SUCCESS,
+  REGISTER_FAIL
+} from "./types";
+import axios from "axios";
+import { returnErrors } from "./errorActions";
+
+export const loadUser = () => (dispatch, getState) => {
+  dispatch({ type: USER_LOADING });
+
+  axios
+    .get("http://localhost:5000/api/user", tokenConfig(getState))
+    .then(res => res.json())
+    .then(res => dispatch({
+      type: USER_LOADED,
+      payload: res.data
+    }))
+    .catch(err => {
+      dispatch(returnErrors(err.response.data, err.response.status));
+      dispatch({
+        type: AUTH_ERROR
+      })
+    })
+}
+
+export const register = ({ username, email, password, graduationYear }) => (dispatch) => {
+  const config = {
+    headers: {
+      "Content-Type": "application/json"
+    }
+  };
+  const body = JSON.stringify({ username, email, password, graduationYear });
+  console.log(body);
+  console.log(process.env.REACT_APP_API_REGISTER)
+  axios
+    .post(process.env.REACT_APP_API_REGISTER, body, config)
+    .then(res => {
+      dispatch({
+        type: REGISTER_SUCCESS,
+        payload: res.data
+      });
+      console.log("post works");
+    })
+    .catch(err => {
+      dispatch(
+        returnErrors(err.response.data, err.response.status, "REGISTER_FAIL")
+      );
+      dispatch({
+        type: REGISTER_FAIL
+      })
+    });
+
+}
+
+export const login = ({ username, password }) => (dispatch) => {
+  const config = {
+    headers: {
+      "Content-Type": "application/json"
+    }
+  };
+  const body = JSON.stringify({ username, password });
+
+  axios
+    .post(process.env.REACT_APP_API_LOGIN, body, config)
+    .then(res =>
+      dispatch({
+        type: LOGIN_SUCCESS,
+        payload: res.data
+      })
+    )
+    .catch(err => {
+      dispatch(
+        returnErrors(err.response.data, err.response.status, "LOGIN_FAIL")
```

Comment:
Here, "LOGIN_FAIL" is the id, right?
Also, it is a special-valued string used across files. So, in order to reduce errors, declare a const variable somewhere, and use that across the files. (The `LOGIN_FAIL` variable is already declared in the types file. You can directly use that)